--- configure.ac_bck	2013-04-02 14:24:49.000000000 -0500
+++ configure.ac	2013-04-02 14:24:05.000000000 -0500
@@ -54,10 +54,8 @@
 AC_CHECK_HEADERS(iomanip.h limits.h values.h float.h)
 AC_CHECK_HEADERS(siginfo.h bits/sigset.h bstring.h sys/types.h sys/select.h)
 AC_CHECK_HEADERS(X11/extensions/transovl.h X11/extensions/readdisplay.h)
-AC_CHECK_HEADERS(X11/extensions/multibuf.h)
-AC_CHECK_HEADERS(sys/filio.h sys/mman.h libc.h)
-AC_CHECK_HEADERS(X11/extensions/multibuf.h)
 AC_CHECK_HEADERS(sys/filio.h sys/mman.h libc.h)
+AC_CHECK_HEADERS([X11/Xlib.h X11/extensions/multibuf.h], [], [], [#include <X11/Xlib.h>])
 #-----------------------------------------------------------------------------
 # Posix threads library is required
 #-----------------------------------------------------------------------------
@@ -554,7 +552,7 @@
       if test "x${HAVE_FTGL_INC}" = "xyes"; then
         AC_MSG_CHECKING([for FTGLTextureFont in -lftgl])
         CPPFLAGS="${FTGL_INCLUDES} ${HAVE_FTGL_UPPERCASE} ${CSF_FREETYPE_INCLUDES} ${CPPFLAGS_old}"
-        AC_TRY_LINK(
+        AC_TRY_COMPILE(
         [
           #ifdef HAVE_FTGL_UPPERCASE
           #include <FTGL/FTGL.h>
